soc/arm/nxp_lpc/lpc55xxx/soc.c

```diff
@@ -67,50 +82,93 @@ static ALWAYS_INLINE void clock_init(void)
 #if defined(CONFIG_SOC_LPC55S06) || defined(CONFIG_SOC_LPC55S16) || \
 	defined(CONFIG_SOC_LPC55S28) || defined(CONFIG_SOC_LPC55S36) || \
 	defined(CONFIG_SOC_LPC55S69_CPU0)
-    /*!< Set up the clock sources */
-    /*!< Configure FRO192M */
-	/*!< Ensure FRO is on  */
+	/* Set up the clock sources */
+	/* Configure FRO192M */
+	/* Ensure FRO is on  */
 	POWER_DisablePD(kPDRUNCFG_PD_FRO192M);
-	/*!< Set up FRO to the 12 MHz, just for sure */
+	/* Set up FRO to the 12 MHz, to ensure we can change the clock freq */
 	CLOCK_SetupFROClocking(12000000U);
-	/*!< Switch to FRO 12MHz first to ensure we can change the clock */
+	/* Switch to FRO 12MHz first to ensure we can change the clock */
 	CLOCK_AttachClk(kFRO12M_to_MAIN_CLK);
 
+	/* Configure XTAL32M */
+#if !defined(CONFIG_SOC_LPC55S36)
+	POWER_DisablePD(kPDRUNCFG_PD_XTAL32M);
+	POWER_DisablePD(kPDRUNCFG_PD_LDOXO32M);
+#endif
+	CLOCK_SetupExtClocking(16000000U);
```

Comment:
Why are these two Power_DisablePD() calls needed here?  When CLOCK_SetupExtClocking() is called below, it also appears to do the same thing.  See https://github.com/nxp-mcuxpresso/mcux-sdk/blob/bd677546404372cdea18a08a5b04232adefd5e9f/devices/LPC55S69/drivers/fsl_clock.c#L403

Reply:
I had noticed that the LPC55S36 was using the values XTALHF and LDOXTALHF, instead of XTAL32M and LDOXO32M that was being set in SetupExtClocking, after a second look I realize that these values are pointing to the same register offset so I will make this change, sorry for the confusion